packages/esm-patient-programs-app/src/programs/delete-program-modal.test.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { deleteProgramEnrollment, useEnrollments } from './programs.resource';
+import DeleteProgramModal from './delete-program.modal';
+import { showSnackbar, getCoreTranslation } from '@openmrs/esm-framework';
+
+jest.mock('./programs.resource', () => ({
+  deleteProgramEnrollment: jest.fn(),
+  useEnrollments: jest.fn(),
+}));
+jest.mock('@openmrs/esm-framework', () => ({
+  showSnackbar: jest.fn(),
+  getCoreTranslation: jest.fn((key, defaultText) => defaultText),
+}));
+jest.mock('react-i18next', () => ({
+  useTranslation: () => ({ t: (key, defaultText) => defaultText }),
+}));
+
+const mockMutateEnrollments = jest.fn();
+(useEnrollments as jest.Mock).mockImplementation(() => ({ mutateEnrollments: mockMutateEnrollments }));
```

Comment:
Please see the correct way to mock functions where we prefer to use `jest.mocked` where possible - https://o3-docs.openmrs.org/docs/frontend-modules/unit-and-integration-testing#annotating-mocks-with-type-information